infrastructure-playbooks/shrink-osd.yml

```diff
@@ -109,14 +114,32 @@
         CEPH_CONTAINER_IMAGE: "{{ ceph_docker_registry + '/' + ceph_docker_image + ':' + ceph_docker_image_tag if containerized_deployment | bool else None }}"
         CEPH_CONTAINER_BINARY: "{{ container_binary }}"
       register: _lvm_list_data
-      delegate_to: "{{ item.0 }}"
-      loop: "{{ _osd_hosts }}"
+      delegate_to: "{{ item }}"
+      loop: "{{ host_list }}"
 
     - name: set_fact _lvm_list
       set_fact:
         _lvm_list: "{{ _lvm_list | default({}) | combine(item.stdout | from_json) }}"
       with_items: "{{ _lvm_list_data.results }}"
 
+    - name: refresh /etc/ceph/osd files non containerized_deployment
+      ceph_volume_simple_scan:
+        cluster: "{{ cluster }}"
+        force: true
+      environment:
+        CEPH_CONTAINER_IMAGE: "{{ ceph_docker_registry + '/' + ceph_docker_image + ':' + ceph_docker_image_tag if containerized_deployment | bool else None }}"
+        CEPH_CONTAINER_BINARY: "{{ container_binary }}"
+      delegate_to: "{{ item }}"
+      loop: "{{ host_list }}"
+      when: not containerized_deployment | bool
+
+    - name: refresh /etc/ceph/osd files containerized_deployment
+      command: "ceph-volume simple scan --force /var/lib/ceph/osd/ceph-{{ item.2 }}"
```

Comment:
this command won't work in containerized deployment, you must run it from within a container..